custom_components/hochwasserportal/api.py

```diff
@@ -285,10 +300,11 @@ def parse_NW(self):
             if data[0]["data"][-1][1] > 0:
                 self.level = data[0]["data"][-1][1]
                 # Get data for stages
-                data_stages = self.fetch_json(
+                data_stages = self.get_json(
                     "https://hochwasserportal.nrw/lanuv/data/internet/stations/100/"
                     + self.ident[3:]
-                    + "/S/alarmlevel.json"
+                    + "/S/alarmlevel.json",
+                    ttl=86400
                 )
                 # List to store water level measurements for specific ts_names
                 water_level_measurements = []
```

Comment:
Isn't it better to move the generation of water_level_measurements to _parse_init_NW()_ and store only the three required water levels in a simple array? Name proposal _self.alarmlevel_.
Then we don't need cachetools at all.